core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileSystem.java

```diff
@@ -19,49 +19,30 @@
 import alluxio.conf.AlluxioConfiguration;
 import alluxio.exception.AlluxioException;
 import alluxio.grpc.OpenFilePOptions;
-import alluxio.metrics.MetricKey;
-import alluxio.metrics.MetricsSystem;
 
-import com.codahale.metrics.Counter;
+import com.google.common.base.Preconditions;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.util.Optional;
 
 /**
  * A FileSystem implementation with a local cache.
  */
 public class LocalCacheFileSystem extends DelegatingFileSystem {
   private static final Logger LOG = LoggerFactory.getLogger(LocalCacheFileSystem.class);
-  private static Optional<CacheManager> sCacheManager;
-
+  private final CacheManager mCacheManager;
   private final AlluxioConfiguration mConf;
 
-  // TODO(binfan): Remove the IOException throwing or make this into a factory method
   /**
+   * @param cacheManage cache manager
    * @param fs a FileSystem instance to query on local cache miss
    * @param conf the configuration, only respected for the first call
    */
-  @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(
-      value = "ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD",
-      justification = "write to static is made threadsafe")
-  public LocalCacheFileSystem(FileSystem fs, AlluxioConfiguration conf) throws IOException {
+  public LocalCacheFileSystem(CacheManager cacheManage, FileSystem fs, AlluxioConfiguration conf) {
     super(fs);
-    // TODO(feng): support multiple cache managers
-    if (sCacheManager == null) {
-      synchronized (LocalCacheFileSystem.class) {
-        if (sCacheManager == null) {
-          try {
-            sCacheManager = Optional.of(CacheManager.create(conf));
-          } catch (IOException e) {
-            Metrics.CREATE_ERRORS.inc();
-            throw new IOException("Failed to create CacheManager", e);
-          }
-        }
-      }
-    }
-    mConf = conf;
+    mCacheManager = Preconditions.checkNotNull(cacheManage, "cacheManager");
```

Comment:
Should this be allowed to be null? It looks like we have fallback logic in the appropriate places

Reply:
no more relevant